### ISSUE TITLE
Expand Margin is not working in Styles for TextEdit

### ISSUE THREAD

**Issue by azagaya** [godotengine/godot#74445]
### Godot version

4.1.dev (5dccc94)

### System information

Ubuntu 22.04, Vulkan renderer, gforce 1650ti

### Issue description

Modifying the Expand Margin property in a style applied to a TextEdit does not draw extending outside the rect like in other controls, which causes the style to be clipped.

### Steps to reproduce

1. Add a TextEdit
2. Add a Style to it
3. Use expand margins
4. Note that it doesn't work as expected

### Minimal reproduction project

[TestTheme.zip](https://github.com/godotengine/godot/files/10892436/TestTheme.zip)

**Comment by azagaya** [comment]
In line 514 of text_edit.cpp we have this line which causes the issue:

```gdscript
RenderingServer::get_singleton()->canvas_item_set_clip(get_canvas_item(), false);
```
Is it safe to remove it? I dont see why this was added

**Comment by AThousandShips** [comment]
Why *shouldn't* it clip the contents?

**Comment by azagaya** [comment]
> Why _shouldn't_ it clip the contents?

Did you try the example project? Is definitely not working as expected, at least not as all the rest of the controls work. The expand margin is useless for TextEdit as it is now.

Btw, this is not "clip the text content", the text is clipped anyways with or without that line removed. This is just clipping the draw rect, which causes nothing to be drawn outside the control rects, which causes expand margin not be usable.

**Comment by azagaya** [comment]
To be honest, i think when fixing these it should be also added that you could move the scroll bars. They are annoying sometimes that overlap with the borders of the theme.

**Comment by RPicster** [comment]
> 

It would be the only control (to my knowledge) with this behaviour:
![image](https://user-images.githubusercontent.com/9423774/222980743-23a702b2-703a-4f86-888b-60d06013e92b.png)

**Comment by azagaya** [comment]
Ok, text is clipping without that line, but not correctly. There must be a way to correctly clip text without this clipping the theme.

**Comment by t-karcher** [comment]
> It would be the only control (to my knowledge) with this behaviour

Nope. ScrollContainer has the same issue, see #78523 (I originally reported the missing drop shadow, but expand margin also doesn't work.)

**Comment by kitbdev** [comment]
- Clip contents can be disabled as of #89993

With clip contents disabled, the expand margin can be seen.

![image](https://github.com/godotengine/godot/assets/10054226/1c7ca371-4a3a-45b4-8520-131aa151fcb6)
